src/dbus_mpris.rs

```diff
@@ -459,7 +419,128 @@ async fn create_dbus_server(api_token: RspotifyToken, spirc: Arc<Spirc>, device_
         }),
     );
 
-    // run forever
-    futures::future::pending::<()>().await;
-    unreachable!();
+    loop {
+        let event = event_rx
+            .recv()
+            .await
+            .expect("Changed track channel was unexpectedly closed");
+        let mut changed_properties: HashMap<String, Variant<Box<dyn RefArg>>> = HashMap::new();
+        let track_id = match event {
+            PlayerEvent::Changed { new_track_id, .. } => Some(new_track_id),
+            PlayerEvent::Started { track_id, .. } => {
```

Comment:
The spec for the Position property says that
> The org.freedesktop.DBus.Properties.PropertiesChanged signal is **not** emitted when this property changes.

So only the Seeked signal should be emitted in this case and not the PropertiesChanged signal. I can implement the Seeked signal to this pull request too if there's interest in that.

Reply:
Yeah, you are right, sorry.

I personally would really like it, if the `Seeked` signal was emitted properly, but this is definitely not a hard requirement for this PR.

Thanks for your work!

Reply:
I just got around to try your PR and noticed that the first part of my comment regarding `Started` vs. `Playing` is actually still valid. If I pause the playback the signal is emitted correctly, continuing the playback however does nothing:

![screenshot](https://user-images.githubusercontent.com/59307989/145711766-f84e10db-dfc3-4925-ba5d-fcc068b3c417.png)

Reply:
Hm, you are right. So I will have to use the `Playing` event. But it would be hard to detect if only the position changed or also something else. In other words it's hard to detect if a Seeked or a PropertiesChanged signal should be emitted. Do you think it would be fine to always emit PropertiesChanged when getting a `Playing` event?

Reply:
For a usual mpris client, I don't see any problem sending those additional `PropertiesChanged` updates. It might still be better to only send them if they are actually needed, to reduce the amount of Spotify API calls. Would it be possible to store the `track_id` outside of the loop and then compare it with the potentially new value or am I missing something that prevents this?

Reply:
Ah, yes. That's definitely possible. I don't know what I was thinking.

Reply:
Thanks! (I can't resolve this thread so you will have to do it :grinning:)